x-pack/plugins/search_indices/public/components/details_page/details_page.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiPageSection, EuiSpacer, EuiButton, EuiPageTemplate } from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { FunctionComponent } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useIndex } from '../../hooks/api/use_index';
+import { i18n } from '@kbn/i18n';
+
+export const SearchIndexDetailsPage: FunctionComponent<
+  RouteComponentProps<{ indexName: string }>
+> = ({ location: { search } }) => {
```

Comment:
There are several ways to type this, so this isn't wrong. I just don't see it as often. A similar type definition that is equivalent would be using `React.FC`:
```suggestion
export const SearchIndexDetailsPage: React.FC<
  RouteComponentProps<{ indexName: string }>
> = ({ location: { search } }) => {
```

But if we don't need to accept children components then what is more common is just defining the props like this:
```suggestion
export type SearchIndexDetailsPageProps = RouteComponentProps<{ indexName: string }>;
export const SearchIndexDetailsPage = ({ location: { search } }: SearchIndexDetailsPageProps) => {
```

See [this blog](https://medium.com/raccoons-group/why-you-probably-shouldnt-use-react-fc-to-type-your-react-components-37ca1243dd13) for more info.